### PR TITLE
add azure object storage backend

### PR DESCRIPTION
**Pull request by icewind1991**
- [x] requires https://github.com/nextcloud/3rdparty/pull/95

Example config:

```php
  'objectstore' =>
  array (
    'class' => 'OC\\Files\\ObjectStore\\Azure',
    'arguments' =>
    array (
      'account_name' => 'accoutname',
      'account_key' => '*********',
      'container' => 'test',
    ),
  ),

```

### CODE REVIEW

**Comment by rullzer** [comment]
3rdparty conflicts.
Also I'm a bit scared to just merge this without tests (which are of course hard to do for azure :S)

**Comment by icewind1991** [comment]
rebased.

Also not a fan of not having tests, luckly the code is minimal with the sdk doing the real work

File: lib/private/Files/ObjectStore/Azure.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+		if (!$this->blobClient) {
+			$connectionString = "DefaultEndpointsProtocol=https;AccountName=" . $this->accountName . ";AccountKey=" . $this->accountKey;
+			$this->blobClient = BlobRestProxy::createBlobService($connectionString);
+		}
+		return $this->blobClient;
+	}
+
+	/**
+	 * @return string the container or bucket name where objects are stored
+	 * @since 7.0.0
```
Comment (MorrisJobke) [inline lib/private/Files/ObjectStore/Azure.php]: 14 I guess 😆

**Review by rullzer — APPROVED** [review]
:elephant:

**Comment by MariusBluem** [comment]
Is there a way to test Azure without MS Azure-Platform? Like AWS S3 can be tested with Minio 🤔

**Comment by rullzer** [comment]
Not as far as I know

**Comment by icewind1991** [comment]
Found https://github.com/azure/azurite, working on getting ci working with it

**Comment by icewind1991** [comment]
tests are here and seem happy

**Review by MorrisJobke — APPROVED** [review]
Code makes sense 👍

**Pull request merged**